go/mysql/datetime/datetime.go

```diff
@@ -778,3 +778,14 @@ func NewDateTimeFromStd(t time.Time) DateTime {
 		Time: NewTimeFromStd(t),
 	}
 }
+
+func NewDateTimeFromSeconds(seconds decimal.Decimal) DateTime {
+	secondsInt64, _ := seconds.Int64()
```

Comment:
I think we should make sure to also include sub-second precision here. I would expect a general conversion function to honor that.

Reply:
Already reverted :)

Reply:
Decided to never reverse-engineer a function value; always compute forward.